encoding/protobuf/encodingIntegration_test.go

```diff
@@ -0,0 +1,189 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package protobuf_test
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/encoding/protobuf"
+	"go.uber.org/yarpc/encoding/protobuf/internal/testpb"
+	"go.uber.org/yarpc/transport/grpc"
+	"go.uber.org/yarpc/yarpcerrors"
+	"go.uber.org/yarpc/yarpctest"
+)
+
+type testStructure struct {
+	name   string
+	req    *yarpctest.Call
+	expReq map[string]string
+}
+
+var allTests map[string]testStructure
+
+func validateReq(testname string, ctx context.Context) (bool, string) {
+	test := allTests[testname]
+	call := yarpc.CallFromContext(ctx)
+	for name, value := range test.expReq {
+		switch name {
+		case "CallerProcedure":
+			if call.CallerProcedure() != value {
+				err := "CallerProcedure '" + call.CallerProcedure() + "' does match with expected value '" + value + "'"
+				return false, err
+			}
+		case "Procedure":
+			if call.Procedure() != value {
+				err := "Procedure '" + call.Procedure() + "' does match with expected value '" + value + "'"
+				return false, err
+			}
+		}
+	}
+	//fmt.Println("Entered in validateReq :  ", testname)
+	return true, ""
+}
+
+func runTest(t *testing.T, test testStructure, client testpb.TestYARPCClient) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	ctx = yarpctest.ContextWithCall(ctx, test.req)
+	defer cancel()
+
+	_, err := client.Unary(ctx, &testpb.TestMessage{Value: test.name})
+	require.NoError(t, err, "unexpected call error")
+}
+
+func TestProtobufEncoding(t *testing.T) {
+	client, cleanup := initalizeClientAndServer(t, &encodingIntegrationTestServer{})
+
+	defer cleanup()
+
+	tests := []testStructure{
+		{
+			name: "test1",
+			req: &yarpctest.Call{
+				Procedure: "ABC1",
+			},
+			expReq: map[string]string{
+				"CallerProcedure": "ABC1",
+				"Procedure":       "uber.yarpc.encoding.protobuf.Test::Unary",
+			},
+		},
+		{
+			name: "test2",
+			req:  &yarpctest.Call{},
+			expReq: map[string]string{
+				"CallerProcedure": "",
+				"Procedure":       "uber.yarpc.encoding.protobuf.Test::Unary",
+			},
+		},
+	}
+	allTests = make(map[string]testStructure)
+	for _, test := range tests {
+		allTests[test.name] = test
+		runTest(t, test, client)
+	}
+
+}
+
+func initalizeClientAndServer(t *testing.T, server testpb.TestYARPCServer) (
+	client testpb.TestYARPCClient,
+	cleanup func(),
+) {
+	serverAddr, cleanupServer := CreateServer(t, server)
+	client, cleanupClient := CreateClient(t, serverAddr)
+	return client, func() {
+		cleanupServer()
+		cleanupClient()
+	}
+}
+
+func CreateServer(t *testing.T, server testpb.TestYARPCServer) (addr string, cleanup func()) {
```

Comment:
nit: this method need not be exposed outside, lower-case the first letter `createServer`